### PR TITLE
[pydeck] Update jupyter-widget to work with JupyterLab 3

### PR DESCRIPTION
**Pull request by dakoop**
For #6989

#### Change List
- Make changes for JupyterLab 3 as suggested by [ipywidgets](https://ipywidgets.readthedocs.io/en/stable/changelog.html#updates-for-widget-maintainers)
- Bump the linked version of deck.gl to 8.7
- Update pyproject.toml to add header

#### Notes
- Not sure if this should be based on the 8.7 release branch or master. Should be easy to change.
- This does not create a prebuilt extension (see <https://github.com/dakoop/deck.gl/tree/pydeck-8.7-prebuilt> for work on that)

### CODE REVIEW

File: modules/jupyter-widget/package.json
Comment (kylebarron) [inline modules/jupyter-widget/package.json:39]: Is there a reason to remove `^1`? Or, is it known which version of jupyter notebook started to move to `^2`? Is this suggested by one of the current dev references?

**Review by kylebarron — APPROVED** [review]

**Comment by kylebarron** [comment]
> * Not sure if this should be based on the 8.7 release branch or master. Should be easy to change

@Pessimistress I would think this should go into master; thoughts?

**Comment by ajduberstein** [comment]
What's the case for not merging this into master?

@dakoop before merging let me test that this works locally.

**Comment by kylebarron** [comment]
> What's the case for not merging this into master?

I think it would better to be merged into master and then cherry picked onto the 8.7 release branch. (`@deck.gl/jupyter-widget` would need another release for these changes to take effect in published pydeck)

**Comment by dakoop** [comment]
Re: branches, I wasn't sure how the version updates happen. Since pydeck ties itself to a particular deck.gl release in `pydeck/frontend_semver.py` (currently `"~8.5.*"`), wasn't sure which version to set if it goes to master and figured this was clear on `8.7-release`. The changes should be the same on any recent branch.

**Comment by Pessimistress** [comment]
This should go into master and then cherry-picked to the 8.7-release branch. There shouldn't be any conflict.

File: modules/jupyter-widget/package.json
Comment (dakoop) [reply modules/jupyter-widget/package.json:39]: The [webpage](https://ipywidgets.readthedocs.io/en/stable/changelog.html#updates-for-widget-maintainers) says

> Update the @jupyter-widgets/base dependency version to include ^4 to work in JupyterLab 3.0. For example, if you had a dependency on @jupyter-widgets/base version ^2 || ^3, update to ^2 || ^3 || ^4 for your widget to work in classic Jupyter Notebook, JupyterLab 1, JupyterLab 2, and JupyterLab 3. See [#2472](https://github.com/jupyter-widgets/ipywidgets/pull/2472) for background.

However, it seems like keeping ^1.1.10 would be fine.

Comment (kylebarron) [reply modules/jupyter-widget/package.json:39]: I'd prefer to keep `^1.1.10` as well unless there's a conflicgt.

**Comment by kylebarron** [comment]
@Pessimistress did you mean to move the base branch back to `8.7-release`?

**Comment by Pessimistress** [comment]
The current PR is done on top of 8.7-release. The release branch cannot be merged  cleanly into master.

**Comment by kylebarron** [comment]
@dakoop Can you make a PR for this off of the `master` branch?

**Comment by dakoop** [comment]
Yes, will do

**Comment by dakoop** [comment]
Ok, I recall more about why I went with 8.7. `frontend_semver.py` is not automatically kept in sync with the current jupyter-widget version. This means that if I check in a change to master which ties pydeck to, e.g., 8.8.0-alpha.6, this reference breaks the next time jupyter-widget's version gets bumped. More in #7025

**Comment by dakoop** [comment]
Replaced by #7026

**Pull request closed without merging**
